### PR TITLE
Update the build matrix

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,6 @@ env:
   global:
     - NOKOGIRI_USE_SYSTEM_LIBRARIES=true
 rvm:
-  - 2.1
```
Comment (cbeer) [inline .travis.yml]: Are we dropping support for 2.1? I seem to remember that 2.1 may be the latest ruby available through some package managers?

File: Gemfile [hunk 2]
```diff
@@ -4,7 +4,7 @@ source 'https://rubygems.org'
 gemspec
 
 gem 'slop', '~> 3.6' if RUBY_PLATFORM == "java"
-gem 'byebug'
+gem 'byebug', platforms: [ :mri_20, :mri_21, :mri_22, :mri_23]
```
Comment (cbeer) [inline Gemfile]: Could this simpy be `:mri`?

File: .travis.yml (hunk 1 above)
Comment (jcoyne) [inline .travis.yml]: We had the most recent two rubies here before, so I just replaced with the most recent two rubies.  If we want to support 3 versions back, that's okay too.

Comment (jcoyne) [inline .travis.yml]: I've re-added 2.1

**Pull request merged**